probe-rs/src/vendor/nxp/mod.rs

```diff
@@ -25,8 +25,8 @@ impl Vendor for Nxp {
             DebugSequence::Arm(MIMXRT10xx::create())
         } else if chip.name.starts_with("MIMXRT11") {
             DebugSequence::Arm(MIMXRT11xx::create())
-        } else if chip.name.starts_with("MIMXRT5") {
-            DebugSequence::Arm(MIMXRT5xxS::create())
+        } else if chip.name.starts_with("MIMXRT5") || chip.name.starts_with("MIMXRT6") {
+            DebugSequence::Arm(MIMXRT5xxS::create(chip))
```

Comment:
Please decode and pass in the chip family here, there is no need to implement the logic in two halves.

Reply:
flash reset procedure is different between `MIMXRT6` and `MIMXRT5`. I need something within `MIMRXRT5xxS` to differentiate between the two. Apart from flash reset, the sequences are exactly the same, so I opted to avoid code duplication.

Do you have a better suggestion as to how to handle the difference?

Reply:
Thanks, the new change is exactly what I had in mind.